bindings/node.js/kzg.cxx

```diff
@@ -152,190 +152,215 @@ Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
+// verifyBlobKzgProof: (blob: Blob, commitmentBytes: Bytes48, proofBytes: Bytes48, setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyBlobKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
 
-    // Extract commitment from parameter
-    Napi::Value commitment = commitments_param[blob_index];
-    auto commitment_bytes = commitment.As<Napi::Uint8Array>().Data();
-    memcpy(&commitments[blob_index], commitment_bytes, BYTES_PER_COMMITMENT);
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 4;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
-  bool verification_result;
-  ret = verify_aggregate_kzg_proof(
-    &verification_result,
-    blobs,
-    commitments,
-    blobs_count,
+  auto blob_bytes = extract_byte_array_from_param(info, 0, "blob");
+  auto commitment_bytes = extract_byte_array_from_param(info, 1, "commitmentBytes");
+  auto proof_bytes = extract_byte_array_from_param(info, 2, "proofBytes");
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
+
+  if (env.IsExceptionPending()) {
+    return env.Null();
+  }
+
+  bool out;
+  C_KZG_RET ret = verify_blob_kzg_proof(
+    &out,
+    (Blob *)blob_bytes,
+    (Bytes48 *)commitment_bytes,
     (Bytes48 *)proof_bytes,
     kzg_settings
   );
 
-  free(commitments);
-  free(blobs);
-
   if (ret != C_KZG_OK) {
-    Napi::Error::New(
-      env,
-      "verify_aggregate_kzg_proof failed with error code: " + std::to_string(ret)
-    ).ThrowAsJavaScriptException();
+    Napi::TypeError::New(env, "Error in verifyBlobKzgProof").ThrowAsJavaScriptException();
     return env.Null();
   }
 
-  return Napi::Boolean::New(env, verification_result);
+  return Napi::Boolean::New(env, out);
 }
 
-// verifyKzgProof: (commitmentBytes: Bytes48, zBytes: Bytes32, yBytes: Bytes32, proofBytes: Bytes48, setupHandle: SetupHandle) => boolean;
-Napi::Value VerifyKzgProof(const Napi::CallbackInfo& info) {
+// verifyBlobKzgProofBatch: (blobs: Blob[], commitmentsBytes: Bytes48[], proofsBytes: Bytes48[], setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyBlobKzgProofBatch(const Napi::CallbackInfo& info) {
   auto env = info.Env();
 
   size_t argument_count = info.Length();
-  size_t expected_argument_count = 5;
+  size_t expected_argument_count = 4;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
-  auto commitment_bytes = extract_byte_array_from_param(info, 0, "commitmentBytes");
-  auto z_bytes = extract_byte_array_from_param(info, 1, "zBytes");
-  auto y_bytes = extract_byte_array_from_param(info, 2, "yBytes");
-  auto proof_bytes = extract_byte_array_from_param(info, 3, "proofBytes");
-  auto kzg_settings = info[4].As<Napi::External<KZGSettings>>().Data();
+  auto blobs_param = info[0].As<Napi::Array>();
```

Comment:
Do we need any checks for these casts to `Napi::Array` here? Looking at extract_byte_array_from_param() we are doing a bunch of checks that are not done here.

Reply:
I don't think so, these are checked in typescript. Let's look into it more later.